pkg/volume/util/operationexecutor/operation_generator.go

```diff
@@ -207,8 +207,28 @@ func (og *operationGenerator) GenerateBulkVolumeVerifyFunc(
 				newAttacherErr)
 			return nil
 		}
-		bulkVolumeVerifier, ok := volumeAttacher.(volume.BulkVolumeVerifier)
 
+		bulkVolumeStatus, ok := volumeAttacher.(volume.BulkVolumeStatus)
+		if ok {
+			volumes, bulkStatusErr := bulkVolumeStatus.GetBulkVolumeStatus(volumeSpecMap)
+			if bulkStatusErr != nil {
```

Comment:
The volumespec map is created from actual state of the world and hence it is not going to fix dangling volumes problem.

Reply:
Hmmm ... I wanted to get "all relevant volumes" - I guess I should merge in the expected state of the world volumes.

(And actually I wanted to do all volumes, but it's hard to get unique volume names for them.)

Reply:
I think we will have to create separate verification mechanism by reading volumes from desired state of the world rather than actual state of world.  After that we will have to poll with cloudprovider if they attached. If yes - then we will have to add them to actual state of the world. 

But again, this will not be perfect. Because - a dangling volume will only be fixed then, when *someone* creates a new pod with same volume.  Also - a dangling volume will never be correctly deleted because it will not be present in either actual state or desired state of world.